lib/process_settings/testing/monitor.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'active_support/core_ext'
+require 'process_settings/monitor'
+
+module ProcessSettings
+  module Testing
+
+    # A special instance of the monitor specifically used for testing that
+    # allows the providing of a settings array from memory to initialize
+    # the ProcessSetting Monitor for testing
+    #
+    # @param Array settings_array
+    # @param Logger logger
+    class Monitor < ::ProcessSettings::Monitor
```

Comment:
We don't want `ProcessSettings::Monitor` to be both a concrete class and a base class. My plan was to make a new `ProcessSettings::BaseMonitor` that would have the commonality. Like `def []`.